jetty-server/src/main/java/org/eclipse/jetty/server/handler/ErrorHandler.java

```diff
@@ -429,21 +429,35 @@ private void writeErrorPlain(HttpServletRequest request, PrintWriter writer, int
 
     private void writeErrorJson(HttpServletRequest request, PrintWriter writer, int code, String message)
     {
+        Throwable cause = (Throwable)request.getAttribute(Dispatcher.ERROR_EXCEPTION);
+        Object servlet = request.getAttribute(Dispatcher.ERROR_SERVLET_NAME);
+
         writer
             .append("{\n")
-            .append("  url: \"").append(request.getRequestURI()).append("\",\n")
-            .append("  status: \"").append(Integer.toString(code)).append("\",\n")
-            .append("  message: ").append(QuotedStringTokenizer.quote(message)).append(",\n");
-        Object servlet = request.getAttribute(Dispatcher.ERROR_SERVLET_NAME);
+            .append("  \"url\": \"").append(request.getRequestURI()).append("\",\n")
+            .append("  \"status\": \"").append(Integer.toString(code)).append("\",\n")
+            .append("  \"message\": ").append(QuotedStringTokenizer.quote(message));
+
         if (servlet != null)
-            writer.append("servlet: \"").append(servlet.toString()).append("\",\n");
-        Throwable cause = (Throwable)request.getAttribute(Dispatcher.ERROR_EXCEPTION);
+        {
+            writer.append(",\n");
+            writer.append("servlet: \"").append(servlet.toString()).append("\"");
+        }
+        if(cause!=null)
+            writer.append(',');
+
+        writer.append("\n");
+
         int c = 0;
         while (cause != null)
         {
-            writer.append("  cause").append(Integer.toString(c++)).append(": ")
-                .append(QuotedStringTokenizer.quote(cause.toString())).append(",\n");
+            writer.append("  \"cause").append(Integer.toString(c++)).append('"').append(": ")
+                .append(QuotedStringTokenizer.quote(cause.toString()));
             cause = cause.getCause();
+            if(cause!=null)
+            {
+                writer.append(",\n");
+            }
```

Comment:
I quite don't like the approach of concatenating strings to generate the JSON.
I would rather prefer to create a `Map`, do the if/else logic filling the `Map` - this will void even more if/else to take care of the commas and carriage returns and indents.
Once we have the `Map`, a simple loop will generate the JSON without worrying too much about commas etc.
This is a rare case anyway, so one more `Map` is not a big deal and the code will be much cleaner.

Reply:
good idea!